hibernate-core/src/test/java/org/hibernate/orm/test/dialect/unit/CockroachDialectVersionTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.dialect.unit;
+
+import org.hibernate.dialect.CockroachDialect;
+import org.hibernate.dialect.DatabaseVersion;
+
+import org.hibernate.testing.TestForIssue;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * @author Jan Schatteman
+ */
+public class CockroachDialectVersionTest {
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-15511")
+	public void testCockroachDialectVersionParsing() {
+		DatabaseVersion dv;
+
+		String versionString = "CockroachDB CCL v21.2.10 (x86_64-unknown-linux-gnu, built 2022/05/02 17:38:58, go1.16.6)";
+		dv = CockroachDialect.parseVersion( versionString );
```

Comment:
Instead of making that version parsing method public, please test this code by creating instances of `CockroachDialect` with `TestingDialectResolutionInfo`. You'll have to adapt that `TestingDialectResolutionInfo` a bit though by introducing a field `String databaseVersion` which should be assigned to what `getDatabaseVersion()` returns now by default, but a new `forDatabaseInfo` variant will additionally take the String explicitly.

Reply:
Not sure if my approach is what you had in mind, let me know your thoughts